src/libstd/io/stdio.rs

```diff
@@ -872,6 +874,93 @@ impl fmt::Debug for StderrLock<'_> {
     }
 }
 
+/// Equivalent to the [`input`] method except that a prompt
+/// is printed before capturing input. This function returns
+/// either success ([`Ok`]) or failure ([`Err`]) containing
+/// the input.
+///
+/// # Note
+///
+/// This function is not the recommended solution to get user input, it should
```

Comment:
I think "not recommended" is a bit strong here. If this method works for your use case, you should definitely use it! Let's maybe just remove up to "If you require more explicit control"?

Reply:
trimmed notes